components/lsp/LSP.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { useState, useEffect, FC } from "react";
+import { ethers } from "ethers";
+
+import LSPForm from "./LSPForm";
+
+import createLSPContractInstance from "./createLSPContractInstance";
+import createERC20ContractInstance from "./createERC20ContractInstance";
+
+import { useConnection } from "../../hooks";
+
+import { Synth } from "../../utils/umaApi";
+export enum ContractState {
+  Open,
+  ExpiredPriceRequested,
+  ExpiredPriceReceived,
+}
+
+interface Props {
+  contractAddress: string;
+  collateralSymbol: string;
+  data: Synth<{ type: "lsp" }>;
+  longTokenBalance: ethers.BigNumber;
+  shortTokenBalance: ethers.BigNumber;
+  refetchLongTokenBalance: () => void;
+  refetchShortTokenBalance: () => void;
+  collateralBalance: ethers.BigNumber;
+  setCollateralBalance: React.Dispatch<React.SetStateAction<ethers.BigNumber>>;
+}
+
+const toBN = ethers.BigNumber.from;
+const LSP: FC<Props> = ({
+  data,
+  longTokenBalance,
+  shortTokenBalance,
+  refetchLongTokenBalance,
+  refetchShortTokenBalance,
+  collateralBalance,
+  setCollateralBalance,
+}) => {
+  console.log("data", data);
+  const { account = "", signer, provider } = useConnection();
+
+  const [lspContract, setLSPContract] = useState<ethers.Contract | null>(null);
+  const [collateralERC20Contract, setCollateralERC20Contract] =
+    useState<ethers.Contract | null>(null);
+
+  const [contractState, setContractState] = useState<ContractState>(
+    data.contractState
+  );
+
+  const [contractExpirationTime, setContractExpirationTime] =
+    useState<string>("");
+
+  const [currentTime, setCurrentTime] = useState<string>("");
+  const [showSettle, setShowSettle] = useState(false);
+
+  // Check if contract is settable.
+  useEffect(() => {
+    if (signer && data.contractState === ContractState.ExpiredPriceRequested) {
+      const lc = createLSPContractInstance(signer, data.address);
+      try {
+        lc.callStatic
```

Comment:
the try catch wont work here because theres no await on this, you probably want to just chain a catch since you arent in an async block

Reply:
Fair, added catch block.